x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutDataFrameAnalyticsAction.java

```diff
@@ -46,12 +71,60 @@ protected void doExecute(Task task, PutDataFrameAnalyticsAction.Request request,
             listener.onFailure(LicenseUtils.newComplianceException(XPackField.MACHINE_LEARNING));
             return;
         }
-
+        
         validateConfig(request.getConfig());
-        configProvider.put(request.getConfig(), ActionListener.wrap(
-            indexResponse -> listener.onResponse(new PutDataFrameAnalyticsAction.Response(request.getConfig())),
-            listener::onFailure
-        ));
+        if (licenseState.isAuthAllowed()) {
+            final String username = securityContext.getUser().principal();
+            RoleDescriptor.IndicesPrivileges sourceIndexPrivileges = RoleDescriptor.IndicesPrivileges.builder()
+                .indices(request.getConfig().getSource())
+                .privileges(SearchAction.NAME)
+                .build();
+            RoleDescriptor.IndicesPrivileges destIndexPrivileges = RoleDescriptor.IndicesPrivileges.builder()
+                .indices(request.getConfig().getDest())
+                 // CreateIndexAction.NAME is more expansive than and create_index, in this instance we want at least create_index.
```

Comment:
Are you sure?  Looking at the definition in `IndexPrivilege`, it appears that `create_index` is simply an alias for `CreateIndexAction.NAME`:

```
    private static final Automaton CREATE_INDEX_AUTOMATON = patterns(CreateIndexAction.NAME);
```

However, rather than change `create_index` to `CreateIndexAction.NAME` it might be better to go the other way with the others.  I think the user also needs the `ReindexAction.NAME` action, and the `write` index privilege would encompass that, `IndexAction.NAME` and `PutMappingAction.NAME` (which we probably need too).

So maybe go with:

```
    .privileges("read", "write", "create_index")
```

Reply:
@droberts195 I am positive that `create_index` != `CreateIndexAction.NAME`, the security team has confirmed this and I have through testing. Initially I thought it was a bug, but apparently its a known thing and has been this way for a while. I would assume through intuition that it behaves the same as the other simplified names, but alas, it does not. 

I am fine with "going whole hog" on the simplified names :).